src/python/pants/base/hash_utils.py

```diff
@@ -62,7 +66,16 @@ def _maybe_encode_dict_key(self, key_obj):
     else:
       return self.encode(key_obj)
 
+  def _is_primitive(self, o):
+    return isinstance(o, (type(None), bool, int, list, text_type, binary_type)
```

Comment:
Wait I don't think that `list` should be here, should it? A list is not primitive, but a collection of primitives. Although looking at line 112 maybe this is correct. 

Perhaps add a `NB` about why list is primitive because this is very surprising to me.

Reply:
Done :)